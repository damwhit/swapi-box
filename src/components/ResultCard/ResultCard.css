.card--favorite-false .btn {
  background-color: red;
}

.card--favorite-true .btn {
  background-color: green;
}
