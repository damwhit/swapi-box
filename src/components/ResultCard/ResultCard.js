import React from 'react';
import './ResultCard.css';

function ResultCard(props) {
  let result = props.value;
  const className = `card card--favorite-${result.isFavorite}`
  return (
    <article
      className={className}
    >
      {result.name}
      <button
        className="btn"
        onClick={props.onClick}
      >
        favorite
      </button>
    </article>
  );
}

export default ResultCard;
