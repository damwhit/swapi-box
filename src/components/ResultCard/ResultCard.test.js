import React from 'react';
import { mount, shallow } from 'enzyme';
import ResultCard from './ResultCard';

describe('ResultCard', () => {

  const favoriteResult = {
    name: 'Chewy',
    isFavorite: true,
  };

  const wrapper = shallow(
    <ResultCard
      value={favoriteResult}
      onClick={jest.fn()}
    />
  );

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should have a className of "starred" if is starred', () => {
    expect(wrapper.is('.card--favorite-true')).toEqual(true);
  });

  it('should call the onClick prop when clicked', () => {
    const onClickMock = jest.fn();

    const wrapper = mount(
      <ResultCard
        value={favoriteResult}
        onClick={onClickMock}
      />
    );

    wrapper.find('.btn').simulate('click');

    expect(onClickMock).toBeCalled();
  }); 
});
