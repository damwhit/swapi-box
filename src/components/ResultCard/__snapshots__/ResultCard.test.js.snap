// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultCard
    onClick={[MockFunction]}
    value={
      Object {
        "isFavorite": true,
        "name": "Chewy",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Chewy",
        <button
          className="btn"
          onClick={[MockFunction]}
        >
          favorite
        </button>,
      ],
      "className": "card card--favorite-true",
    },
    "ref": null,
    "rendered": Array [
      "Chewy",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorite",
          "className": "btn",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "favorite",
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Chewy",
          <button
            className="btn"
            onClick={[MockFunction]}
          >
            favorite
          </button>,
        ],
        "className": "card card--favorite-true",
      },
      "ref": null,
      "rendered": Array [
        "Chewy",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorite",
            "className": "btn",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "favorite",
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
